import-export-cli/cmd/mg/delete.go

```diff
@@ -0,0 +1,50 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package mg
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+)
+
+const deleteCmdLiteral = "delete"
+const deleteCmdShortDesc = "Delete an API in Microgateway"
+const deleteCmdLongDesc = `Delete an API by specifying name, version, host, username 
+and optionally vhost by specifying the flags (--name (-n), --version (-v), --host (-c), 
+--username (-u), and optionally --vhost (-t). Note: The password can be included 
+via the flag --password (-p) or entered at the prompt.`
+
+const deleteCmdExamples = utils.ProjectName + ` ` + mgCmdLiteral + ` ` + deleteAPICmdLiteral +
+	` -n petstore -v 0.0.1 --host https://localhost:9095 -u admin -t www.pets.com`
```

Comment:
```suggestion
const deleteCmdExamples = utils.ProjectName + ` ` + mgCmdLiteral + ` ` + deleteAPICmdLiteral +
	` -n petstore -v 0.0.1 --host https://localhost:9095 -u admin -t www.pets.com
	NOTE: The 4 flags (--name (-n), --version (-v), --host (-c) and --username (-u)) are mandatory. (The password can be included via the flag --password (-p) or entered at the prompt)`
```

Reply:
Shall we change this like here? We do not need to mention specifically about the optional flags.

Also, I was wondering why the username is mandatory and password is not. Can't we make both optional?
You can refer to "apictl login" command to get an idea [1]. In there we have made both the username and password optional. Shall we change here similar to "apictl login"?
@uvindra and @npamudika WDYT?

[1] https://apim.docs.wso2.com/en/latest/learn/api-controller/getting-started-with-wso2-api-controller/#login-to-an-environment

Reply:
yes +1, we can do the same for username as well. If the username if not provided we can promt and ask maintaining the consistency

Reply:
Seems there might be a login, and both username and password might only be accepted during then. I'll update accordingly when it is finalized.

Reply:
great. Please incorporate the necessary changes to the mgw deploy command too when the login is added.

Reply:
Will do.